### PR TITLE
Add remove image link in admin.

### PR DESCRIPTION
**Pull request by glogiotatidis**
Allow snippet editors to remove images added in image fields.

### CODE REVIEW

**Comment by glogiotatidis** [comment]
Hey @ScottDowne, do you have time to review this one?

**Comment by glogiotatidis** [comment]
Ping @ScottDowne r?

**Comment by ScottDowne** [comment]
Hey, yeah, I do. I saw this go by while on vacation then forgot.

First question while I get started is what's the difference between this pr and #162?

File: snippets/base/static/js/templateDataWidget.js [hunk 1]
```diff
@@ -161,7 +168,10 @@
          */
         onImageFieldChange: function(input) {
             var self = this;
+            var preview = $(input).siblings('img')[0];
```
Comment (ScottDowne) [inline snippets/base/static/js/templateDataWidget.js]: This var is created already later in this function.

File: snippets/base/static/templates/snippetDataWidget.html [hunk 2]
```diff
@@ -12,6 +12,7 @@
         {% elif variable.type == types.image %}
           <img src="{{ originalData[variable.name] }}">
           <input type="file" class="image-input">
+          <a href="#" class="image-input-remove">Remove image</a>
```
Comment (ScottDowne) [inline snippets/base/static/templates/snippetDataWidget.html]: Just a nit, probably doesn't really matter:

If easy to add a 5px left margin/padding to make it line up with the browse button?

**Comment by ScottDowne** [comment]
@glogiotatidis that's all I could find. Looks good.

**Comment by ScottDowne** [comment]
I also noticed, but was unrelated to this patch is a console error while loading the snippet editor:

TypeError: data is undefined: templateDataWidget.js:88:25

Also displaying hiding an image element when the image src is empty is easy, instead of displaying a broken link: `img[src=""] {
    display: none;
}` or something.

File: snippets/base/static/js/templateDataWidget.js (hunk 1 above)
Comment (glogiotatidis) [inline snippets/base/static/js/templateDataWidget.js]: I moved it from line 183 up here. Do I miss something?

Comment (ScottDowne) [inline snippets/base/static/js/templateDataWidget.js]: That should do it!

File: snippets/base/static/templates/snippetDataWidget.html (hunk 2 above)
Comment (glogiotatidis) [inline snippets/base/static/templates/snippetDataWidget.html]: Good point, added in #162

**Comment by glogiotatidis** [comment]
> I also noticed, but was unrelated to this patch is a console error while loading the snippet editor:
> 
> TypeError: data is undefined: templateDataWidget.js:88:25
> 
> Might be just for me, might not be reproducible for you.
> 
> Also hiding an image element when the image src is empty, instead of displaying a broken link, is easy: img[src=""] { display: none; } or something.

Fixed in #162 as well.

**Comment by glogiotatidis** [comment]
> First question while I get started is what's the difference between this pr and #162?

We currently have two diverged branches. On production atm is `legacy` and we're moving soon production to `master` which features a bunch of library upgrades and fixes. I try to bring needed functionality to both branches thus the two identical PRs against the two different branches. 

Beautification comments and other fixes I made in #162 to bring to the new platform. I'll keep this one just for the removeImage functionality.

**Comment by glogiotatidis** [comment]
Thanks for your time @ScottDowne. Do you mind spending some time reviewing #162 that resolves the rest of your review comments?

**Pull request merged**